cloudsync/smartsync.py

```diff
@@ -232,6 +233,9 @@ def _ent_to_smartinfo(self, ent: Optional[SyncEntry], local_dir_info: Optional[D
             mtime = local_dir_info.mtime
             is_synced = True
         else:
+            if ent[LOCAL].exists in (TRASHED, MISSING) \
```

Comment:
This method isn't for filtering, smart_listdir_path is for that. See comment below

Reply:
I elected to put this here because I think we may need a more generic solution than just a filter within listdir. For instance, if a file is trashed remotely I think we should probably return `None` in `smart_info_path()` and `smart_info_oid()` as well. I think another good solution would be to write a separate function for filtering, `_smart_ent_exists(ent)`, or something like that.

Reply:
I think it's important not to trust this utility function with decision making on behalf of the callers. Moving the filtering logic to it's own method is a fine idea

Reply:
That's fair, will move to separate function